build.gradle

```diff
@@ -133,3 +133,8 @@ task pullResolverKeys(type: Download) {
     src 'https://raw.githubusercontent.com/unstoppabledomains/uns/main/resolver-keys.json'
     dest 'src/main/resources/com/unstoppabledomains/config/knownRecords.json'
 }
+
+build {
+    dependsOn pullNetworkConfig
+    dependsOn pullResolverKeys
+}
```

Comment:
This will pull uns contracts and keys during each build making it easier to remember to update configs regularly.

The only downside is, potentially, if a breaking change is introduced in the smart contracts and someone builds the lib locally they will get an updated contract config and may run into some errors. But it shouldn't affect existing builds on jitpack/maven.

@sammyluo @enaqx What do you guys think?

Reply:
Will it break backward compatibility? Say we have it for 6.1.0. Then If the smart contract introduce a new breaking change, and we update and publish 6.2.0. Then apps that uses 6.1.0 will stop working unless they update to 6.2.0, no?

Reply:
It shouldn't. We still have to commit the changed files to the repo. Builds of 6.1 will have the old contracts and will not change. So apps that use jitpack or maven will not be affected. Only if someone tries to checkout 6.1 locally and build it they will run into errors.

And if there is a breaking change in the contracts it will probably be a major version of the lib.

I guess there is no need to force this. However, 99% of our config updates are non-breaking (mostly new resolution records). And this should help keep it up to date. Basically when you are working on the library you will see if the dependencies changed - git will show modified files. So you don't have to remember to pull updates manually.

A different hook might work too but I'm not sure if there's anything that we can "hook" onto for dependency updates. And there is no way to set our uns repo as a dependency since it's not a java lib.